node/test/toolrunnertests.ts

```diff
@@ -410,6 +410,225 @@ describe('Toolrunner Tests', function () {
+                assert.fail('tasklist | FIND "cmd" failed to run: ' + err.message);
+            })
+            .fin(function () {
+                tl.popd();
+                done();
+            })
+        }
+        else {
+            var grep = tl.tool(tl.which('grep', true));
+            grep.arg('ssh');
+
+            var ps = tl.tool(tl.which('ps', true));
+            ps.arg('ax');
+            ps.pipeExecOutputToTool(grep);
+
+            ps.on('stdout', (data) => {
+                output += data.toString();
+            });
+
+            ps.exec(_testExecOptions)
+            .then(function (code) {
+                assert.equal(code, 0, 'return code of exec should be 0');
+                assert(output && output.length > 0 && output.indexOf('ssh') >= 0, 'should have emitted stdout ' + output);
+            })
+            .fail(function (err) {
+                assert.fail('ps ax | grep ssh failed to run: ' + err.message);
+            })
+            .fin(function () {
+                tl.popd();
+                done();
+            })
+        }
+    })
+    it('Exec pipe output to another tool, fails if first tool fails', function(done) {
+        this.timeout(1000);
+
+        tl.pushd(__dirname);
+
+        var _testExecOptions: trm.IExecOptions = {
+            cwd: __dirname,
+            env: {},
+            silent: false,
+            failOnStdErr: false,
+            ignoreReturnCode: false,
+            outStream: testutil.getNullStream(),
+            errStream: testutil.getNullStream()
+        }
+
+        var output = '';
+        var errOut = '';
+        if (os.platform() === 'win32') {
+            var find = tl.tool(tl.which('FIND', true));
+            find.arg('"cmd"');
+
+            var tasklist = tl.tool(tl.which('tasklist', true));
+            tasklist.arg('bad');
+            tasklist.pipeExecOutputToTool(find);
+
+            tasklist.on('stdout', (data) => {
+                output += data.toString();
+            });
+
+            tasklist.on('stderr', (data) => {
+                errOut += data.toString();
+            });
+
+            tasklist.exec(_testExecOptions)
+            .then(function (code) {
+                    assert.fail('tasklist bad | find "cmd" was a bad command and it did not fail');
+                })
+                .fail(function (err) {
+                    assert(errOut && errOut.length > 0 && errOut.indexOf('ERROR: Invalid argument/option') >= 0, 'error output from tasklist command is expected');
+                    assert(err && err.message && err.message.indexOf('tasklist.exe') >=0, 'error from tasklist is not reported');
+                })
+            .fin(function () {
+                tl.popd();
+                done();
+            })
+
+        }
+        else {
+            var grep = tl.tool(tl.which('grep', true));
+            grep.arg('ssh');
+
+            var ps = tl.tool(tl.which('ps', true));
+            ps.arg('bad');
```

Comment:
Can we flatten the if / else code to one shared block of logic?  Seems like we could have variables above on tool names and expected output which are conditionalized on platform and one block of code.